logging_format/tests/test_visitor.py

```diff
@@ -0,0 +1,224 @@
+"""
+Visitor tests.
```

Comment:
There are at least three hooks into `flake8`. You can hook into logical lines, physical lines, or the abstract syntax tree (AST).

I opted for the latter since it meant using a tree traversal instead of regex. This file is full of AST test cases. It doesn't use the `flake8` interface directly.